### PR TITLE
webpack fixes and optimizations

### PR DESCRIPTION
**Pull request by bridiver**
use webpack dev server for about pages
newtab loading is slightly faster in production

### CODE REVIEW

**Comment by bridiver** [comment]
@bbondy

File: app/about-preferences.html [hunk 1]
```diff
@@ -3,19 +3,19 @@
...
-  <script src="gen/aboutPages.entry.js" defer></script>
-  <script src="./ext/l20n.min.js" defer></script>
-</head>
-<body>
-  <div id="appContainer"/>
-</body>
+  <head>
+    <meta charset="UTF-8">
+    <!-- TODO: Refactor away all unsafe-inline content -->
+    <meta http-equiv="Content-Security-Policy" content="default-src 'none'; form-action 'none'; referrer no-referrer; script-src 'self'; connect-src 'self' ws://localhost:8080; style-src 'self' 'unsafe-inline'; font-src 'self'">
```
Comment (bbondy) [inline app/about-preferences.html]: what is ws://localhost:8080?

Comment (bridiver) [inline app/about-preferences.html]: some websocket that webpack uses

File: js/components/frame.js [hunk 2]
```diff
@@ -224,29 +231,23 @@ class Frame extends ImmutableComponent {
...
+      if (this.props.frame.get('location') === 'about:certerror' &&
+          security && security.get('certDetails')) {
+        // Don't send certDetails.cert since it is big and crashes the page
+        this.webview.send(messages.CERT_DETAILS_UPDATED, {
+          url: security.get('certDetails').url,
+          error: security.get('certDetails').error
+        })
+      }
+    })
+    this.webview.addEventListener('did-navigate-in-page', () => {
```
Comment (bbondy) [inline js/components/frame.js]: I'm fairly sure moving this in did-finish-load isn't right. If you allow ads and load slashdot I think this gets called a bunch of times before the frame is actually loaded. WHy was this changed away from did-frame-finish-load w/ isMainFrame check?

Comment (bridiver) [inline js/components/frame.js]: did-finish-load is only called for the main frame
`void WebContents::DidFinishLoad(content::RenderFrameHost* render_frame_host,
                                const GURL& validated_url) {
  bool is_main_frame = !render_frame_host->GetParent();
  Emit("did-frame-finish-load", is_main_frame);

  if (is_main_frame)
    Emit("did-finish-load");
}
`

Comment (bbondy) [inline js/components/frame.js]: Did it change from some fixes in electron? Because it was being called multiple times before.

Comment (bridiver) [inline js/components/frame.js]: not sure, but I'll double check that it only gets called once

Comment (bridiver) [inline js/components/frame.js]: I only see one call per page on slashdot and several others

Comment (bridiver) [inline js/components/frame.js]: looks like some other change already removed the duplicate, but this puts it in the event for the main frame only

**Comment by bridiver** [comment]
newtab load time is now < 30ms

File: app/about-about.html [hunk 3]
```diff
@@ -3,17 +3,23 @@
...
+    <meta http-equiv="Content-Security-Policy" content="default-src 'none'; form-action 'none'; referrer no-referrer; script-src 'self' 'unsafe-inline'; connect-src 'self' ws://localhost:8080">
+    <meta name="availableLanguages" content="en-US">
+    <meta name="defaultLanguage" content="en-US">
+    <title data-l10n-id="aboutPages"></title>
+    <script>
+      window.setTimeout(function() {
+        var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
+        po.src = 'gen/aboutPages.entry.js';
+        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
+      }, 30);
```
Comment (bridiver) [inline app/about-about.html]: seems to work with a slight lower delay, but it can't be arbitrarily low because at some point it reverts back to synchronous loading so I kept it at 30 to be safe

Comment (bridiver) [inline app/about-about.html]: actually a better fix is probably to do this with a dom event. The 30ms probably puts it outside of the load

**Comment by bbondy** [comment]
Looks great, merging!
